### ISSUE TITLE
add a bazel BUILD file

### ISSUE THREAD

**Issue by dominichamon** [google/benchmark#191]
cmake is great, but it can't generate bazel (yet?). Create a bazel BUILD file alongside the cmake stuff.

**Comment by EricWF** [comment]
Any particular reason for bazel? Alternatively we have a pretty simple build process and we could simply provide a buildit script or makefile.

**Comment by dominichamon** [comment]
company allegiance, and ease of reintegration into google's source. ie, having a parallel build setup is overhead that i was trying to avoid by using cmake, but there are benefits to having a bazel build, but they're entirely selfish :)

the benefit of cmake over a script or makefile are two-fold: generation of native "projects" for MSVC users, XCode, etc, and dependency enforcement. as such, i don't think we can have just non-cmake.

i don't actually know if bazel has something similar or relies on all dependencies being integrated into the project so it's hermetic. ie, if we'd need to have submodules/subtrees for any libraries on which we depend. if it does, then maybe a simpler script is the way forward?

**Comment by jupp0r** [comment]
As a starting point, this is what I'm using for [prometheus-cpp](https://github.com/jupp0r/prometheus-cpp) (hard coded to C++11s regex engine, because I require c++11 anyway):

```
cc_library(
    name = "googlebenchmark",
    srcs = glob(["src/*.cc"],
                exclude = ["src/re_posix.cc", "src/gnuregex.cc"]),
    hdrs = glob(["src/*.h", "include/benchmark/*.h"],
                exclude = ["src/re_posix.h", "src/gnuregex.h"]),
    includes = [
         "include",
    ],
    visibility = ["//visibility:public"],
    copts = [
          "-DHAVE_STD_REGEX"
    ],
)
```

**Comment by ChrisCummins** [comment]
Thanks @jupp0r !

**Comment by ffreling** [comment]
I think this issue has been fixed in version v1.4.0
https://github.com/google/benchmark/releases/tag/v1.4.0
